pkg/grafeas/helper.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2021 The Rode Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grafeas
+
+import (
+	"context"
+	"fmt"
+	"github.com/rode/rode/pkg/constants"
+	grafeas_proto "github.com/rode/rode/protodeps/grafeas/proto/v1beta1/grafeas_go_proto"
+	"go.uber.org/zap"
+	"strings"
+)
+
+//go:generate counterfeiter -generate
+
+//counterfeiter:generate . Helper
+type Helper interface {
+	ListVersionedResourceOccurrences(ctx context.Context, resourceUri, pageToken string, pageSize int32) ([]*grafeas_proto.Occurrence, string, error)
+}
+
+type helper struct {
+	logger        *zap.Logger
+	grafeasCommon grafeas_proto.GrafeasV1Beta1Client
+}
+
+func NewHelper(logger *zap.Logger, grafeasCommon grafeas_proto.GrafeasV1Beta1Client) Helper {
+	return &helper{logger, grafeasCommon}
+}
+
+func (h *helper) ListVersionedResourceOccurrences(ctx context.Context, resourceUri, pageToken string, pageSize int32) ([]*grafeas_proto.Occurrence, string, error) {
+	log := h.logger.Named("ListVersionedResourceOccurrences")
+
+	buildOccurrences, err := h.grafeasCommon.ListOccurrences(ctx, &grafeas_proto.ListOccurrencesRequest{
+		Parent:   constants.RodeProjectSlug,
+		PageSize: constants.MaxPageSize,
```

Comment:
This could change to loop through the pages in another PR; although since this call is strictly build occurrences, it would be surprising to have 1000 of them. 

Maybe it should log a warning (or error?) if there's a page token in the response?

Reply:
yeah I guess I should have worded it as a "potential problem".  I agree it seems unlikely for now.  I'll just create an issue for it.

> Maybe it should log a warning

I could also do this, good suggestion.